tuf/client/updater.py

```diff
@@ -1493,47 +1580,58 @@ def _update_metadata_if_changed(self, metadata_role,
     # untrusted data is not decompressed prior to verifying hashes, or
     # decompressing a file that may be invalid or partially intact.
     compression = None
-    compressed_fileinfo = None
 
-    # Extract the fileinfo of the uncompressed version of 'metadata_role'.
-    uncompressed_fileinfo = self.metadata['current'][referenced_metadata] \
-                                         ['meta'] \
-                                         [uncompressed_metadata_filename]
+    # Extract the versioninfo of the uncompressed version of 'metadata_role'.
+    expected_versioninfo = self.metadata['current'][referenced_metadata] \
+                                        ['meta'] \
+                                        [uncompressed_metadata_filename]
 
     # Check for the availability of compressed versions of 'snapshot.json',
     # 'targets.json', and delegated Targets (that also start with 'targets').
     # For 'targets.json' and delegated metadata, 'referenced_metata'
     # should always be 'snapshot'.  'snapshot.json' specifies all roles
-    # provided by a repository, including their file lengths and hashes.
+    # provided by a repository, including their version numbers.
     if metadata_role == 'snapshot' or metadata_role.startswith('targets'):
```

Comment:
Better addressed in https://github.com/vladimir-v-diaz/tuf/commit/ddd06f620507ceae6c6ad783f300610674429efb.